### DISCUSSION TITLE
How to predict a system's energies through a pt file? 🙏

### DISCUSSION THREAD

**Discussion by The-one-on-the-moon** [deepmodeling/dpdata#732] [Q&A] [answered]
I was trying to verify a pt file's accuracy which is finetuned from dpa-2.
Then I met an issue about predicting energies through a pt file.
``` python
training_systems = dpdata.LabeledSystem(path, fmt='deepmd/npy')

# predict = training_systems.predict('./frozen_model.pth')
predict = training_systems.predict('./model.ckpt.pt')

```
The error was below
``` bash
 File "~/deepmd-kit/lib/python3.11/site-packages/deepmd/infer/deep_eval.py", line 189, in _load_graph
    graph_def.ParseFromString(f.read())
google.protobuf.message.DecodeError: Error parsing message with type 'tensorflow.GraphDef'

```
I've done this prediction by a pb file, so I wonder if it works with pt file.

Thanks for your support, looking forward to your replies !

**Comment by njzjz** [comment, marked as answer]
Do you use DeePMD-kit v3?

**Reply by The-one-on-the-moon** [reply]
Yes DeePMD-kit v3.0.0b2

**Reply by njzjz** [reply]
Actually, the code you posted is not v3.0.0b2

>  File "~/deepmd-kit/lib/python3.11/site-packages/deepmd/infer/deep_eval.py", line 189, in _load_graph
>     graph_def.ParseFromString(f.read())

**Reply by The-one-on-the-moon** [reply]
I apologize for my mistake. I checked my environment and realized I was not using v3 as you suggested. After switching to v3, everything worked perfectly. Thanks a lot for your help!
